src/averageAges.js

```diff
@@ -55,7 +86,30 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const arrayWomen = people.filter(item => item.sex === 'f');
+  const peopleAge = [];
+
+  if (onlyWithSon) {
+    for (let i = 0; i < arrayWomen.length; i++) {
+      for (let j = 0; j < people.length; j++) {
+        if (arrayWomen[i].name === people[j].mother && people[j].sex === 'm') {
+          peopleAge.push(people[j].born - arrayWomen[i].born);
+        }
+      }
+    }
+  } else {
+    for (let i = 0; i < arrayWomen.length; i++) {
+      for (let j = 0; j < people.length; j++) {
+        if (arrayWomen[i].name === people[j].mother) {
+          peopleAge.push(people[j].born - arrayWomen[i].born);
+        }
+      }
+    }
+  }
+
+  const averageAge = peopleAge.reduce((sum, age) => sum + age, 0);
```

Comment:
DRY 
![image](https://user-images.githubusercontent.com/70009205/209686533-ef6a3dd1-88e0-4e9c-80e3-f749d90fe202.png)
move repeated code to additional functions